syft/formats/syftjson/decoder.go

```diff
@@ -18,5 +22,26 @@ func decoder(reader io.Reader) (*sbom.SBOM, error) {
 		return nil, fmt.Errorf("unable to decode syft-json: %w", err)
 	}
 
+	if err := outOfDateParser(doc.Schema.Version, internal.JSONSchemaVersion); err != nil {
+		log.Warn(err)
+	}
+
 	return toSyftModel(doc)
 }
+
+func outOfDateParser(documentVerion string, parserVersion string) error {
+	documentV, err := semver.NewVersion(documentVerion)
+	if err != nil {
+		return err
```

Comment:
Maybe it's worth wrapping this err? It might not make sense if it gets logged directly.

Reply:
wrapping would be great to give context to what version couldn't be parsed (where the error would hopefully fill in why)